anvil_schema/schema/mapping_schema_object.json

```
@@ -147,6 +147,7 @@
         {"name": "reference_assembly", "datatype": "string", "array_of": true, "required": false},
         {"name": "file_name", "datatype": "string", "array_of": false, "required": false},
         {"name": "file_ref", "datatype": "fileref", "array_of": false, "required": false},
+        {"name": "supplementary_file_flag", "datatype": "boolean", "array_of": false, "required": false},
```

```suggestion
        {"name": "supplementary_file_flag", "datatype": "boolean", "array_of": false, "required": true},
```

What would `null` mean if we allowed the column to be nullable?

---

Thanks for the quick review! The way we've written the logic to populate this field, it should never be null (but rather should always be either True or False). The only complication with making this field formally "required" is that we are populating it as a secondary process, and making the field required in TDR's purview would result in the initial load of the table failing unless we updated that code to pass a default value. From your standpoint, you can consider this field to be non-nullable.

---

Makes sense.